### PR TITLE
feat: Add audit trail for create, update and delete healthcare profes…

### LINKED ISSUE #275: Add audit trails for healthcare providers functions
# Summary

We want to have audit logs of any changes to data. This way we can see when data is add/edited/deleted. 
Ideally, this should also let us undo/rollback changes in the future.

This issue is for the healthcare provider methods from this list

- facilities (#274)
- healthcare (#275) 
- submissions (#276)

# Requirements

- [ ] Add auditing to all the service methods
- [ ] when deleting, add the record that was deleted into the log
    - [ ] ensure it's in a format that can be used to revert a deletion
- [ ] when updating, record the changes in the log
- [ ] when adding, just point to the record id that was created

Depends on #79

### PR DESCRIPTION
**Pull request by Anissa3005**
Resolves #275 

Add audit trailing to the `healthcareProfessionalService`.
# What changed

1. Changed batch writes to transactions to enable rollbacks
2. Added audit trail for creating a healthcare professional
3. Added audit trail for updating a healthcare professional
4. Added audit trail for deleting a healthcare professional

# Testing instructions

### CODE REVIEW

File: src/services/auditLogService.ts [hunk 1]
```diff
@@ -9,8 +9,8 @@ export async function createAuditLog(
     action: gqlTypes.ActionType, 
     objectType: gqlTypes.ObjectType, 
     updatedBy: string, 
-    newData: Partial<dbSchema.Submission> | gqlTypes.Submission | null,
-    oldData: Partial<dbSchema.Submission> | gqlTypes.Submission | null,
+    newData: Partial<dbSchema.Submission> | gqlTypes.Submission | string | null,
+    oldData: Partial<dbSchema.Submission> | gqlTypes.Submission | string | null,
```
Comment (Anissa3005) [inline src/services/auditLogService.ts:13]: This function needs to be refactored, but wanted to do this in a separate PR since I need to change some things in the submission service if a refactor. I Already created a issue for it.

File: src/services/healthcareProfessionalService.ts [hunk 2]
```diff
@@ -229,34 +231,51 @@ export async function createHealthcareProfessional(
...
+            if (!createdAuditLog.isSuccesful) {
+                throw new Error(`Failed to create and audit log on ${gqlTypes.ActionType.Create}`)
             }
-        }
-
-        //this will update only the fields that are provided and are not undefined.
-        batch.set(healthcareProfessionalRef, newHealthcareProfessional, { merge: true })
+        })
+        
+        // batch.set(healthcareProfessionalRef, newHealthcareProfessional, { merge: true })
```
[the remark this answers is not in the document]
Comment (ermish) [reply src/services/healthcareProfessionalService.ts:276]: I removed it so we can merge!
